src/crystalla/matrix.cr

```diff
@@ -341,6 +341,14 @@ module Crystalla
       s
     end
 
+    # Returns the sum of the diagonal elements. Only useful for numeric matrices.
+    def trace
+      raise ArgumentError.new "Number of rows (#{number_of_rows}) does not match number of columns (#{number_of_cols})" unless square?
+      (0...number_of_cols).inject(0) do |memo, i|
```

Comment:
I'd use `(0...number_of_cols).sum(0.0) { |i| self[i, i] }`. Also please use `0.0` as the starting number, otherwise the result ends up being `Int32 | Float64`